### DISCUSSION TITLE
Is there a way to write modules in rust and integrate them with typescript?

### DISCUSSION THREAD

**Discussion by CoolDeveloper101** [denoland/deno#8874] [Q&A] [answered]
Is there a way to use rust with typescript in deno?

**Comment by ghost** [comment, marked as answer]
Not sure what you meant by "use rust with typescript," but your best bet would be to use the unstable and incomplete native plugin system.

**Reply by CoolDeveloper101** [reply]
I meant something like writing modules in rust and integrating it with typescript?

**Reply by StreakInTheSky** [reply]
> I meant something like writing modules in rust and integrating it with typescript?

You can compile Rust to Web Assembly, and run that from JS.

**Reply by ghost** [reply]
As Rust cannot be integrated with TypeScript at all, it's either compilation to native plugin or WebAssembly.
Can you give a more concrete example of what you're thinking of?
